### PR TITLE
[Tests-Only] Adjust WebDavContext asserts and exceptions for then and given steps

### PR DESCRIPTION
**Pull request by jasson99**
## Description
This PR includes the adjustments in the asserts and exceptions for the then and given steps respectively in the WebDavContext file.

## Related Issue
#36810 

## How Has This Been Tested?
CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 1]
```diff
@@ -1185,6 +1205,12 @@ public function downloadedContentShouldStartWith($start) {
 				)
 			);
 		}
+		Assert::assertEquals(
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebDav.php]: Need to delete the Exception code above.

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 2]
```diff
@@ -2940,7 +2966,10 @@ public function userStoresFileIdForPath($user, $path) {
 	public function userFileShouldHaveStoredId($user, $path) {
 		$currentFileID = $this->getFileIdForPath($user, $path);
 		Assert::assertEquals(
-			$currentFileID, $this->storedFileID
+			$currentFileID,
+			$this->storedFileID,
+			__METHOD__
+			. "User '$user' file '$path' does not have the previously stored id '{$this->storedFileID}', but have '$currentFileID'."
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebDav.php]:
```suggestion
			. " User '$user' file '$path' does not have the previously stored id '{$this->storedFileID}', but has '$currentFileID'."
```

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 3]
```diff
@@ -3011,7 +3040,15 @@ public function propfindResultShouldContainNumEntries($numFiles) {
 		if ($multistatusResults === null) {
 			$multistatusResults = [];
 		}
-		Assert::assertEquals((int) $numFiles, \count($multistatusResults));
+		Assert::assertEquals(
+			(int) $numFiles,
+			\count($multistatusResults),
+			"Expected result to contain '"
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebDav.php]:
```suggestion
			__METHOD__ . " Expected result to contain '"
```

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 4]
```diff
@@ -478,9 +484,12 @@ public function userHasMovedFile(
 		$this->response = $this->makeDavRequest(
 			$user, "MOVE", $fileSource, $headers
 		);
-		Assert::assertEquals(
-			201, $this->response->getStatusCode()
-		);
+		if ($this->response->getStatusCode() !== 201) {
+			throw new Exception(
+				__METHOD__ . "Failed moving resource '$fileSource' to '$fileDestination'.
```
[the remark this answers is not in the document]
Comment (jasson99) [reply tests/acceptance/features/bootstrap/WebDav.php]: fixed

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 5]
```diff
@@ -478,9 +484,12 @@ public function userHasMovedFile(
...
+			Expected status code was '201' but got '" . $this->response->getStatusCode() . "'"
```
[the remark this answers is not in the document]
Comment (jasson99) [reply tests/acceptance/features/bootstrap/WebDav.php]: fixed

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 6]
```diff
@@ -749,7 +758,8 @@ public function userUsingPasswordShouldNotBeAbleToDownloadFile(
 			400, $this->getResponse()->getStatusCode(), 'download must fail'
```
[the remark this answers is not in the document]
Comment (jasson99) [reply tests/acceptance/features/bootstrap/WebDav.php]: fixed

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 7]
```diff
@@ -749,7 +758,8 @@ public function userUsingPasswordShouldNotBeAbleToDownloadFile(
...
 		);
 		Assert::assertLessThanOrEqual(
-			499, $this->getResponse()->getStatusCode(), '4xx error expected'
+			499, $this->getResponse()->getStatusCode(), '4xx error expected but got status code "'
```
[the remark this answers is not in the document]
Comment (jasson99) [reply tests/acceptance/features/bootstrap/WebDav.php]: fixed

File: tests/acceptance/features/bootstrap/WebDav.php (hunk 1 above)
Comment (jasson99) [reply tests/acceptance/features/bootstrap/WebDav.php]: fixed

File: tests/acceptance/features/bootstrap/WebDav.php (hunk 2 above)
Comment (jasson99) [reply tests/acceptance/features/bootstrap/WebDav.php]: fixed

File: tests/acceptance/features/bootstrap/WebDav.php (hunk 3 above)
Comment (jasson99) [reply tests/acceptance/features/bootstrap/WebDav.php]: fixed

**Review by phil-davis — APPROVED** [review]

**Pull request merged**